src/components/color-picker/color-picker.ts

```diff
@@ -188,6 +201,13 @@ export default class SlColorPicker extends ShoelaceElement implements ShoelaceFo
     this.removeEventListener('focusout', this.handleFocusOut);
   }
 
+  firstUpdated() {
+    // we have to wait for SlInput's validity object to be initialized
+    setTimeout(() => {
```

Comment:
This shouldn't be required since `firstUpdated` runs after render. Using `setTimeout` is generally unreliable for things like this. If there's something else being setup that we need to wait for, we should make it more explicit.

Reply:
I think, the problem is, that `firstUpdated` of the inner `sl-input` will be called AFTER `firstUpdated` of `sl-color-picker` is called (please see [this demo](https://codesandbox.io/s/color-picker-issue-ts-dzs7ju?file=/src/index.ts)).
This and the fact that quite some SL methods like for example `checkValidity`, `reportValidity`, `setCustomValidity` and now `get validity` and `get validationMessage` are not safe if called before the corresponding components have been connected (=> `document.createElement('sl-input').checkValidity()` will throw an error ... same with `document.createElement('sl-range').stepUp()` etc.). 

I will make `get validity` for `sl-input` safe to be called at any time. But the others I will leave as is, as I consider this a general challenge which is not in the scope of this task, I think.

Reply:
Correct, that's because the color picker renders the input so the color picker initializes first, then the input. You can use `input.updateComplete.then(…)` as one way of ensuring it's been updated. That will be more reliable than `setTimeout()`.

Another lesser known property of Lit is [`hasUpdated`](https://lit.dev/docs/components/lifecycle/#hasupdated), which will be true once the component has updated at least once.

I hope that helps!

> This and the fact that quite some SL methods like for example checkValidity, reportValidity, setCustomValidity and now get validity and get validationMessage are not safe if called before the corresponding components have been first updated (=> document.createElement('sl-input').checkValidity() will throw an error ... same with document.createElement('sl-range').stepUp() etc.).
>
> I'll fix this single issue in SlColorPicker. But the others I will leave as is, as I consider this a general challenge which is not in the scope of this task, I think.

Nobody should be doing that. Custom elements have to be registered and usually connected/initialized before their DOM will be fully available. That's just the nature of the beast, even outside of Lit.